internal/auth/tokenloader_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+// TestCachedFileTokenLoader_Basic covers basic cache and reload logic for the token loader.
+func TestCachedFileTokenLoader_Basic(t *testing.T) {
+	tmpfile, err := os.CreateTemp(t.TempDir(), "tokenfile-*.txt")
+	require.NoError(t, err)
+	defer os.Remove(tmpfile.Name())
+
+	token := "my-secret-token"
+	_, err = tmpfile.WriteString(token + "\n")
+	require.NoError(t, err)
+	tmpfile.Close()
+
+	loader := cachedFileTokenLoader(tmpfile.Name(), 100*time.Millisecond)
+
+	// First load - should read from file
+	t1, err := loader()
+	require.NoError(t, err)
+	assert.Equal(t, token, t1, "expected %q, got %q")
+
+	// Change the file, but within cache interval
```

Comment:
there is no way to guarantee that in CI environment this will be within the cache interval. You need to test with a mock time.Now function and advance the time manually.